focolab/image-processing
# Wbu seg migration

### Purpose ###
This is the first piece of an effort spanning four repos to centralize template matching code. These repos are:
1. https://github.com/focolab/wb-utils: current source of authority
2. https://github.com/focolab/image-processing: future (centralized) source of authority
3. https://github.com/focolab/gcamp-extractor: client which currently duplicates code from `wb-utils` and depends on `image-processing`; should import currently duplicated code from `image-processing` instead
4. https://github.com/focolab/bfd-core: hosts an example which currently imports template matching code from `wb-utils` to do peakfinding; should import from `image-processing` instead

Work items to complete this migration are as follows:
1. Refactor currently duplicated code and move it into `image-processing` (this change)
2. Update `wb-utils` to remove currently duplicated code and call refactored version in `image-processing` instead
3. Update `gcamp-extractor` to remove currently duplicated code and call refactored version in `image-processing` instead
4. Update `wb-utils` to remove currently duplicated code and call refactored version in `image-processing` instead

### Testing ###
1. Update `gcamp-extractor` locally to use code included in this change:
    1. Code in https://github.com/focolab/gcamp-extractor/commit/94155c53e0b5cda86aaec31e0dcc68c217610512
    2. Update import line in `gcamp-extractor's` `setup.py` to `improc @ git+https://github.com/focolab/image-processing@wbu_seg_migration` and reinstall
2. Verify that `example.py` in `gcamp-extractor` runs without errors
